### ISSUE TITLE
terraform.tfstate file not updated after importing an existing resouce

### ISSUE THREAD

**Issue by jinpenglv** [hashicorp/terraform#29148]
I already have an existing Azure storage account and a blob container `test2021`. Then I used `terraform` to add this resource only. As expected, when I ran `terraform apply`, an error was popped out:

**_A resource with the ID \<my-resource-id-path\> already exists - to be managed via Terraform this resource needs to be imported into the State_**

So I imported it first using:

`terraform import <terraform_resource_path> <my-resource-id-path>`

Then I got:

_**Import successful!
The resources that were imported are shown above. These resources are now in your Terraform state and will henceforth be managed by Terraform.**_

But when I checked the `terraform.tfstate`, nothing is updated, the blob container `test2021` is not there.  So the import of an existing resource is not successful, which indicates a bug. 

Spec:
Terraform v1.0.2
on darwin_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.67.0

Let me know if you need anything else to further help debug/fix the issue. 

Thanks

**Comment by jbardin** [comment]
Hi @jinpenglv,

Thanks for filing the issue, but we will need some information to try and determine what is happening here. It would help if you could provide a way to reproduce the issue, including the configuration for terraform and the individual resource in question. If you can't create a standalone reproduction case, we could start with the trace logs from the import command, which may indicate if there are any problems happening with the import itself.

**Comment by jbardin** [comment]
Hi @mskucherawy, what you are describing was fixed before the v1.0 release, but if you have an example which fails with the latest release that would be a great help.

**Comment by RomanTi** [comment]
Details to the comment left by @mskucherawy: 
I made a typo in the name of the remote object to import and terraform told me:

**_Import successful!_** 

But in fact it didn't change the state file and didn't import anything. 
I was importing `aws_route53_record` resource. 

When I fixed the typo it was able to import it properly.

I saw this behavior on TF 0.15.3. On 0.14.5 I got the proper error:

**_Error: Cannot import non-existent remote object_**
